buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy

```diff
@@ -65,6 +65,9 @@ class PrecommitTasks {
             precommitTasks.add(configureLoggerUsage(project))
         }
 
+        if (project.path.startsWith(":x-pack:plugin:")) {
```

Comment:
This check does not belong under `buildSrc`, which should be unaware of the individual projects. Instead, the task should be installed in the `build.gradle` of the corresponding x-pack directory or could also be packaged as a plugin.

Reply:
I pushed commits that do this (in x-pack/plugin/build.gradle, I did not package this as a plugin and I do not intend to at this time).